File: Lab.Technical.Exercise/Lab.Technical.Exercise.WebApp/ClientApp/src/app/core/models/album.ts
export interface IArtist {
    id: number,
    name: string
}

export interface ILabel {
    id: number,
    name: string
}

export interface IStock {
    id: number,
    quantity: number
}

export interface IAlbum {
    id: number;
    name: string;
    artist: IArtist;
    label: ILabel;
    type: string;
    stock: IStock;
}

export interface ICreateAlbumRequest{
    albumName: string;
    artistId: number;
    labelId: number;
    albumType: number;
    stock: number;
}

export interface IAlbumFormOptionsData{
    artists: IArtist[],
    labels: ILabel[]
}

export const AlbumType = new Map<number, string>([
    [111, 'CD'],
    [222, 'Vinyl']
  ]);

File: Lab.Technical.Exercise/Lab.Technical.Exercise.DataContracts/Album.cs
using Lab.Technical.Exercise.DataContracts.Enums;

namespace Lab.Technical.Exercise.DataContracts
{
    public class Album
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public Artist Artist { get; set; }
        public Label Label { get; set; }
        public AlbumType Type { get; set; }
        public Stock Stock { get; set; }
    }
}